orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -134,35 +108,16 @@ def _create_output_dim_names(output, output_idx, from_sequence):
                             " Compute will continue, but unexpected results may occur!")
 
         sample_outputs = model_copy(*sample_inputs_copy, **sample_kwargs_copy)
-        sample_output_type = type(sample_outputs)
-        if isinstance(sample_outputs, torch.Tensor):
-            output_names, output_dynamic_axes = _create_output_dim_names(sample_outputs, 0, False)
-        elif isinstance(sample_outputs, abc.Mapping):
-            output_names, output_dynamic_axes = _create_output_dim_names_from_mapping(sample_outputs)
-        elif isinstance(sample_outputs, abc.Sequence):
-            for idx, out in enumerate(sample_outputs):
-                tmp_output_names, tmp_output_dynamic_axes = _create_output_dim_names(out, idx, True)
-                output_names += tmp_output_names
-                output_dynamic_axes.update(tmp_output_dynamic_axes)
-        else:
-            raise TypeError('ORTModule does not support the following model output type {}'.format(type(sample_outputs)))
+
+        # Parse the output and populate the output_names and output_dynamic_axes to be used for onnx export
+        output_idx = [0]
+        _ortmodule_output_transformation.parse_outputs_and_extract_names_and_dynamic_axes(
```

Comment:
Recursion did go a long way here. The only little thing is having outputs being passed as arguments as opposed to return types. Do you think we can refactor a bit to make outputs being returned instead?

Reply:
yeah sure, I think that should be doable.